crates/project/src/project.rs

```diff
@@ -11692,7 +11695,45 @@ fn include_text(server: &lsp::LanguageServer) -> bool {
         .unwrap_or(false)
 }
 
-async fn load_shell_environment(dir: &Path) -> Result<HashMap<String, String>> {
+async fn load_direnv_environment(dir: &Path) -> Result<Option<HashMap<String, String>>> {
+    let Ok(direnv_path) = which::which("direnv") else {
+        return Ok(None);
+    };
+
+    let direnv_output = smol::process::Command::new(direnv_path)
+        .args(["export", "json"])
+        .current_dir(dir)
+        .output()
+        .await
+        .context("failed to spawn direnv to get local environment variables")?;
+
+    anyhow::ensure!(
+        direnv_output.status.success(),
+        "direnv exited with error {:?}",
+        direnv_output.status
+    );
+
+    let output = String::from_utf8_lossy(&direnv_output.stdout);
+    if output.is_empty() {
+        return Ok(None);
+    }
+
+    Ok(Some(
+        serde_json::from_str(&output).context("failed to parse direnv output")?,
+    ))
+}
+
+async fn load_shell_environment(
+    dir: &Path,
+    load_direnv: &DirenvSettings,
+) -> Result<HashMap<String, String>> {
+    // Start by loading the direnv environment
+    let direnv_environment = match load_direnv {
+        DirenvSettings::ShellHook => None,
```

Comment:
It will get loaded anyways, right? By `cd`ing into the directory. As in: I think the setting suggests an either-or decision, when it fact it's more that `ShellHook` is the default and `direct` is an additional thing, no?

Reply:
If the direct option is selected, then when the shell hook is invoked, direnv would do nothing because it adds a marker environment variable to indicate that is has been activated

Reply:
There's a comment on line 11744 — can you update that to include this bit here?

Reply:
Sure!